malariagen_data/ag3.py

```diff
@@ -2489,6 +2490,92 @@ def aa_allele_frequencies(
 
         return df_aaf
 
+    def plot_frequencies_heatmap(
+        self, df, index, index_name, max_len=100, width=500, colorbar=True
+    ):
+
+        """Generate a Plotly.express frequency heatmap using pandas DataFrame output from ag3.snp_allele_frequencies()
+        or ag3.gene_cnv_frequencies() that has been user-filtered to just rows of interest (fewer rows than max_len).
+
+        Parameters
+        ----------
+        df : pandas DataFrame
+           Filtered output from ag3.snp_allele_frequencies() or ag3.gene_cnv_frequencies.
+        index : list of str
+            A list of one or more column headers present in the input dataframe, these become
+            the heatmap y-axis row labels. When taken together, these columns must produce a unique index.
+        index_name : str
+            This is the y-axis label that will be displayed on the heatmap
+        max_len : int, optional
+            Displaying large styled dataframes may cause ipython notebooks to crash.
+        width : int
+            Width of styled output.
+        colorbar : bool, optional
+            If False, colorbar is not output.
+
+        Returns
+        -------
+        Plotly.express heatmap built from the index and "frq_" columns of input DataFrame, where
+        increasing frequency is denoted by intensity of red coloration.
+
+        """
+        # check len of input
+        assert len(df) <= max_len
```

Comment:
Suggest either raise a `ValueError` here with a helpful message, or slice the dataframe to the first `max_len` rows.